editor.planx.uk/src/@planx/components/PlanningConstraints/Modal.tsx

```diff
@@ -0,0 +1,247 @@
+  entities: Constraint["data"] | null;
+  metadata?: Metadata;
+  inaccurateConstraints: InaccurateConstraints;
+  setInaccurateConstraints: (
+    value: React.SetStateAction<InaccurateConstraints>,
+  ) => void;
+}
+
+const ERROR_MESSAGES = {
+  checklist: "Select at least one option",
+  input: "Enter a value",
+};
+
+export const OverrideEntitiesModal = ({
+  showModal,
+  setShowModal,
+  fn,
+  entities,
+  metadata,
+  inaccurateConstraints,
+  setInaccurateConstraints,
+}: OverrideEntitiesModalProps) => {
+  const [checkedOptions, setCheckedOptions] = useState<string[] | undefined>(
+    inaccurateConstraints?.[fn]?.["entities"],
+  );
+  const [showChecklistError, setShowChecklistError] = useState<boolean>(false);
+  const [textInput, setTextInput] = useState<string | undefined>(
+    inaccurateConstraints?.[fn]?.["reason"],
+  );
+  const [showInputError, setShowInputError] = useState<boolean>(false);
+
+  const title = `Which ${
+    metadata?.plural?.toLowerCase() || "entities"
+  } are inaccurate?`;
+
+  const closeModal = (_event: any, reason?: string) => {
+    if (reason && reason == "backdropClick") {
+      return;
+    }
+
+    // Clear any non-submitted inputs on cancel & sync parent state
+    setCheckedOptions(undefined);
+    setTextInput(undefined);
+    const newInaccurateConstraints = omit(inaccurateConstraints, fn);
+    setInaccurateConstraints(newInaccurateConstraints);
+
+    // Close modal
+    setShowModal(false);
+  };
+
+  const changeCheckbox =
+    (id: string) =>
+    (_checked: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined) => {
+      let newCheckedIds;
+
+      if (checkedOptions?.includes(id)) {
+        newCheckedIds = checkedOptions.filter((e) => e !== id);
+      } else {
+        newCheckedIds = [...(checkedOptions || []), id];
+      }
+
+      if (newCheckedIds.length > 0) {
+        setShowChecklistError(false);
+      }
+      setCheckedOptions(newCheckedIds);
+    };
+
+  const changeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value.length > 1) {
+      setShowInputError(false);
+    }
+    setTextInput(e.target.value);
+  };
+
+  const validateAndSubmit = () => {
+    const invalidChecklist = !checkedOptions || checkedOptions.length === 0;
+    const invalidInput = !textInput || textInput.trim().length === 0;
+
+    // All form fields are required to submit
+    if (invalidChecklist && invalidInput) {
+      setShowChecklistError(true);
+      setShowInputError(true);
+    } else if (invalidChecklist) {
+      setShowChecklistError(true);
+    } else if (invalidInput) {
+      setShowInputError(true);
+    } else {
+      // Update inaccurateConstraints in parent component state on valid submit
+      const newInaccurateConstraints = {
+        ...inaccurateConstraints,
+        ...{ [fn]: { entities: checkedOptions, reason: textInput } },
+      };
+      setInaccurateConstraints(newInaccurateConstraints);
+      setShowModal(false);
+    }
+  };
```

Comment:
I've written my own simple validation here to start (similar to `UploadAndLabel` modal), but perhaps a good case to formalise using Formik/yup later down the line.

Reply:
Ah just seen a11y as a follow up listen in PR description sorry!

Reply:
Updated outermost modal `Box` to be `component="form"` and added `type="submit"` to button :+1: 

Agree Formik is still one to come back to!